### PR TITLE
added qualifier.py for Heiner

### PR DESCRIPTION
**Pull request by IgNoRaNt23**
Added my qualifier.py

### CODE REVIEW

File: heiner_qualifier/qualifier.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    def __lt__(self, other):
+        return self.publication_date < other.publication_date
+
+    def short_introduction(self,n_characters) -> str:
+        return self.content[:n_characters+1].rsplit(maxsplit=1)[0]
+
+    def most_common_words(self, n_words):
+
+        clean_list = [word.lower() for word in re.split(r'[^A-Za-z]', self.content) if word != '']
```
Comment (MushroomMaula) [inline heiner_qualifier/qualifier.py:89]: Empty word are false by the defaul so you can just check
```py
if word
```

File: heiner_qualifier/qualifier.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        count_dict = {}
+
+        for word in clean_list:
+            try:
+                count_dict[word] += 1
+            except KeyError:
+                count_dict[word] = 1
```
Comment (MushroomMaula) [inline heiner_qualifier/qualifier.py:97]: A more pythonic way would be to use the `collections.defaultdict`.

```py
count_dict = defaultdict(int)  # every no existent key starts at 0
for word in clean_list:
    count_dict[word] += 1
```

File: heiner_qualifier/qualifier.py (hunk 1 above)
Comment (IgNoRaNt23) [reply heiner_qualifier/qualifier.py:89]: Yep, but None, empty lists/dicts/tuples are as well. Explicit is better than implicit!

File: heiner_qualifier/qualifier.py (hunk 2 above)
Comment (IgNoRaNt23) [reply heiner_qualifier/qualifier.py:97]: Let's talk about 'pythonic' stuff in our next meeting. 
Other than that, I like that solution

File: heiner_qualifier/qualifier.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    @content.setter
+    def content(self, value):
+        self.last_edited = datetime.datetime.now()
+        self._content = value
+
+    @classmethod
+    def _get_id(cls):
+        cls._id += 1
+        return cls._id
```
[the remark this answers is not in the document]
Comment (IgNoRaNt23) [reply heiner_qualifier/qualifier.py:73]: While `@classmethod`s are often used for factory methods, that's by far not the only application and they're not only intended for that. If you want classes that inhereit from the Article-class to have their own counter, `@classmethod`s are the most elegant solution

File: heiner_qualifier/qualifier.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    _id = -1
+
+    def __init__(self, title: str, author: str, publication_date: datetime.datetime, content: str):
+        self.title = title
+        self.author = author
+        self.publication_date = publication_date
+        self.content = content
+
+        self.id = self._get_id()
```
[the remark this answers is not in the document]
Comment (IgNoRaNt23) [reply heiner_qualifier/qualifier.py:57]: Se my other comment regarding `@classmethod`s. This version doesn't support inheritance. At least if you dont want to have a counter for every subclass (NewspaperArticle, BlogArticle, MagazineArticel?).

File: heiner_qualifier/qualifier.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+        self.field_type = field_type
+        self.values = {}
+
+    def get_att_name(self, obj):
+        obj_dict = type(obj).__dict__
+        for att in obj_dict.keys():
+            if obj_dict[att] == self:
+                return att
+        raise Exception
```
[the remark this answers is not in the document]
Comment (IgNoRaNt23) [reply heiner_qualifier/qualifier.py:34]: Yep, would have saved a lot if time if I knew that before

File: heiner_qualifier/qualifier.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MushroomMaula) [reply heiner_qualifier/qualifier.py:73]: Thats a good point, I didn't think of inheritance.

File: heiner_qualifier/qualifier.py (hunk 1 above)
Comment (MushroomMaula) [reply heiner_qualifier/qualifier.py:89]: Well naming a variable `word`, makes it pretty obvious its about strings but I guess it comes down to preference

**Pull request merged**